### PR TITLE
Cleanup, adding 'average' bar

### PR DESCRIPTION
**Pull request by DougBanksPersonal**
Adding new "average" bar to graph to show average of current values.
Adding legend to enlarged graph.
General cleanup: 
- using fixed sizes, dynamic scaling gets too crazy.
- splitting TextPanel class, the maximized and non-maximized versions are different enough it's easier to just split.
- got rid of split notion of "stat button display type" vs "stat type".  Made sense when I was worrying about displaying two stats in a single button.  We are not doing that.  Keeping things saner.

### CODE REVIEW

File: CoreScriptsRoot/Modules/Stats/DecoratedValueLabel.lua [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  self._label.TextXAlignment = Enum.TextXAlignment.Left
+  self._label.TextYAlignment = Enum.TextYAlignment.Center
+  
+  StatsUtils.StyleTextWidget(self._label)
+  
+  return self
+end
+  
+function DecoratedValueLabelClass:SetZIndex(zIndex)
+  self._frame.ZIndex = zIndex
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Stats/DecoratedValueLabel.lua]: do any of these overlap? might be a good idea to explicitly set some to a higher zindex if so. By default if zindex is the same we render in child order, but that can always change if mess with parenting items separately.

File: CoreScriptsRoot/Modules/Stats/StatsAggregator.lua [hunk 2]
```diff
@@ -90,7 +90,8 @@ function StatsAggregatorClass:GetAverage()
   for i = 0, self._numSamples-1, 1 do
     retval = retval + self._samples[i]
   end
-  return retval / self._numSamples
+  local average = retval / self._numSamples
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Stats/StatsAggregator.lua]: this change seem superfluous? for readability?

File: CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua [hunk 3]
```diff
@@ -20,46 +20,58 @@ local BarGraphClass = require(CoreGuiService.RobloxGui.Modules.Stats.BarGraph)
...
+  StatsUtils.StyleTextWidget(self._midLabel)
+  StatsUtils.StyleTextWidget(self._bottomLabel)
 
   self:_layoutElements()
 
   return self
 end
 
+function StatsAnnotatedGraphClass:SetZIndex(zIndex)
+  self._frame.ZIndex = zIndex
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua]: same thing here about index

File: CoreScriptsRoot/Modules/Stats/DecoratedValueLabel.lua (hunk 1 above)
Comment (DougBanksPersonal) [inline CoreScriptsRoot/Modules/Stats/DecoratedValueLabel.lua]: They don't overlap with each other.
In general, there's a common pattern of 
Left side has text.
Right side has graph.
We can lay stuff out so each side doesn't have internal overlapping items (nothing on right side runs over anything else on right side).
But the text may go extra long and goosh into the right side, overlapping graph.
Hence the ZIndex functions: once everything is all set up we can set elements in each side to the depth we want.

File: CoreScriptsRoot/Modules/Stats/StatsAggregator.lua (hunk 2 above)
Comment (DougBanksPersonal) [inline CoreScriptsRoot/Modules/Stats/StatsAggregator.lua]: Reverted

File: CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua (hunk 3 above)
Comment (DougBanksPersonal) [inline CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua]: See above.

**Pull request merged**